store/store_test.go

```diff
@@ -20,10 +20,9 @@ func TestStore(t *testing.T) {
 	suite := headertest.NewTestSuite(t)
 
 	ds := sync.MutexWrap(datastore.NewMapDatastore())
-	store, err := NewStoreWithHead(ctx, ds, suite.Head())
-	require.NoError(t, err)
+	store := newStoreWithHead(t, ctx, ds, suite.Head())
```

Comment:
Looks like new constrcutor `newTestStore` can be used here. It also manages lifecycle under the hood and it seems store was not stopped here before.

Reply:
Yes and no :( 

`newStoreWithHead` just creates a store with a given datastore.
`newTestStore` also starts and stops the store.

I don't see a good reason to increasing PR size which will involve big tests rewrite (mostly refactoring but still this will touch every test). So, I thought about that but I thing this can/should be done separately.

WDYT ?

Reply:
In my understanding, single contractor should not require major refactoring, while removing duplicating logic of helper functions. It should just satisfy both:
- rceates a store with a given datastore
- starts and stops the store.

As I mentioned there are tests, that missing store lifecycle management (missing close call), so it will also resolve this issue.

Reply:
Done.